src/app/app.module.ts

```diff
@@ -9,7 +9,7 @@ import { AppComponent } from './app.component';
 
 @NgModule({
   declarations: [AppComponent],
-  imports: [BrowserModule, AppRoutingModule, CoreModule, SharedModule],
+  imports: [BrowserModule, AppRoutingModule, CoreModule.forRoot(), SharedModule.forRoot()],
```

Comment:
Why do we need ```forRoot()``` in CoreModule and SharedModule, and don't - in others?

Reply:
This function, create service is singleton, other modules not provide services.